controllers/workspace/status.go

```diff
@@ -109,6 +81,51 @@ func (r *DevWorkspaceReconciler) updateWorkspaceStatus(workspace *dw.DevWorkspac
 	return reconcileResult, reconcileError
 }
 
+func syncConditions(workspaceStatus *dw.DevWorkspaceStatus, currentStatus *currentStatus) {
+	currTransitionTime := metav1.Time{Time: clock.Now()}
+
+	// Set of conditions already set on the workspace
+	existingConditions := map[dw.DevWorkspaceConditionType]bool{}
+	for idx, workspaceCondition := range workspaceStatus.Conditions {
+		existingConditions[workspaceCondition.Type] = true
+
+		currCondition, ok := currentStatus.conditions[workspaceCondition.Type]
+		if !ok {
+			// Didn't observe this condition this time; set status to unknown
+			workspaceStatus.Conditions[idx].LastTransitionTime = currTransitionTime
+			workspaceStatus.Conditions[idx].Status = corev1.ConditionUnknown
+			workspaceStatus.Conditions[idx].Message = ""
+			continue
+		}
+
+		// Update condition if needed
+		if workspaceCondition.Status != currCondition.Status || workspaceCondition.Message != currCondition.Message {
+			workspaceStatus.Conditions[idx].LastTransitionTime = currTransitionTime
+			workspaceStatus.Conditions[idx].Status = currCondition.Status
+			workspaceStatus.Conditions[idx].Message = currCondition.Message
+		}
+	}
+
+	// Check for conditions we need to add
+	for condType, cond := range currentStatus.conditions {
+		if existingConditions[condType] {
+			// Condition is already present and was updated (if necessary) above
+			continue
+		}
+		workspaceStatus.Conditions = append(workspaceStatus.Conditions, dw.DevWorkspaceCondition{
+			LastTransitionTime: currTransitionTime,
+			Type:               condType,
+			Status:             cond.Status,
+			Message:            cond.Message,
+		})
+	}
+
+	// Sort conditions to avoid unnecessary updates
+	sort.SliceStable(workspaceStatus.Conditions, func(i, j int) bool {
+		return strings.Compare(string(workspaceStatus.Conditions[i].Type), string(workspaceStatus.Conditions[j].Type)) > 0
```

Comment:
Would it be a good UX improvement if you order conditions in the order they are expected to appear?

Reply:
:+1: Done